### ISSUE TITLE
stat call killing performance

### ISSUE THREAD

**Issue by electrofloat** [storj/storj#5757]
**Description**
According to: https://forum.storj.io/t/storage2-exists-check-workers-1/21953/3
`stat()` call is killing performance. Should be enough to just use `access` syscall with `F_OK`, since stat's result is ignored anyway:
https://github.com/storj/storj/blob/41bfbbe772219259c90491fb818927eb39cd78fd/storagenode/piecestore/endpoint.go#L241

**Possible Fix**
 Should be enough to just use `access` syscall with `F_OK`.

**Comment by thepaul** [comment]
I think this may be based on some incorrect assumptions. There isn't any particular reason why `access()` should be noticeably faster than `stat()` on a typical Linux system. Both should get all the info they need from reading the appropriate directory entry. If anything, `access()` is probably written as a wrapper to `stat()`. I'd be surprised if `access()` is faster than `stat()` on Windows too, but there's a lot I don't know going on there. Or maybe SELinux affects access and stat differently?

Basically, I want to know, do we have any profiling info to back this up, and if we do, are we sure it applies to the majority of nodes?

Also, I don't think the Exists endpoint is even being used right now. It was created to ease the load on nodes from a full-system scan we were performing, and that's done now.
